### ISSUE TITLE
IC components act strangly

### ISSUE THREAD

**Issue by Shuudoushi** [MrTJP/ProjectRed#984]
I've noticed that a few of the gates and such act very differently while making ICs then their block counterparts. Namely the transparent latch, which as an IC component has no delay between the input and output when pulsing the lock with a pulse former, causing issues when making something like a shift register (namely the shift register locking up).

(Hopefully this hasn't been posted before, I couldn't find anything while searching, but i may also just be blind or not looking for the right thing.)

**Comment by Shuudoushi** [comment]
Oh, and a video or something can be made just encase my way of explaining is hard to follow.

**Comment by MrTJP** [comment]
Currently, IC gates have to delay. They are not intended to act the same. To simulate the same behavior, place a repeater within the IC on all sides of the transparent latch. (I'm assuming you are just using a TLatch only inside the IC)

Eventually, a minimum 2 tick delay will be enforced.

**Comment by Shuudoushi** [comment]
Ok, that explains a lot. Thanks.
